community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/index/AbstractLuceneIndex.java

```diff
@@ -92,14 +93,33 @@ public void create() throws IOException
      */
     public void open() throws IOException
     {
-        Map<File,Directory> indexDirectories = indexStorage.openIndexDirectories();
-        for ( Map.Entry<File,Directory> indexDirectory : indexDirectories.entrySet() )
+        Stream<Map.Entry<File,Directory>> indexDirectories = indexStorage.openIndexDirectories().entrySet().stream();
+        List<AbstractIndexPartition> list = null;
+        try
         {
-            partitions.add( partitionFactory.createPartition( indexDirectory.getKey(), indexDirectory.getValue() ) );
+            list = indexDirectories.map( e -> createPartition( e.getKey(), e.getValue() ) )
+                                       .collect( Collectors.toList() );
         }
+        catch ( UncheckedIOException e )
```

Comment:
i do not think that map -> and wrapping/catching exceptions should be here since it makes it less readable, it's correct observation that it should use intermediate list instead.